torch/utils/cpp_extension.py

```diff
@@ -1555,7 +1584,11 @@ def _run_ninja_build(build_directory: str, verbose: bool, error_prefix: str) ->
         raise RuntimeError(message) from e
 
 
-def _import_module_from_library(module_name, path, is_python_module):
+def _import_module_from_library(module_name, path, is_python_module, is_standalone):
+    if is_standalone:
+        assert not IS_WINDOWS
```

Comment:
seems better to raise a NotImplementedError, unless the intention here is that the precondition for calling this function is that you're not on Windows (distinguish between "Developer error" and "we just haven't implemented this yet")

Reply:
In this case it was an internal assert. (Earlier code should have prevented that bit from being reached, this just checked that it did.) I put those asserts anywhere that would stop making sense if the top `NotImplementedError` was removed.